File: client/iih_stub/iih.ci_stub/iih.ci.ord/app/d/CiapplissheetAggDO.cs
using System;
using xap.mw.core.data;
using xap.mw.core.meta;

namespace iih.ci.ord.app.d
{
	/// <summary>
	/// 检验申请单DO
	/// </summary>
	public class CiapplissheetAggDO : BaseAggDO
	{	
	    private static string CIAPPLISSHEETDO_CIAPPLISSHEETORDO="iih.ci.ord.app.d.CiAppLisSheetOrDO";

        public CiapplissheetAggDO() {
            this.setParentDO(new CiAppLisSheetDO());
        }

  	    public CiAppLisSheetDO getParentDO() {
		    return ((CiAppLisSheetDO) base.ParentDO);
	    }      
	    
	    public void setParentDO(CiAppLisSheetDO headDO) {
		    base.ParentDO=headDO;
	    }
	    
	    public CiAppLisSheetOrDO[] getCiAppLisSheetOrDO() {		    
		    
		    BaseDO[] dos=getChildren(Type.GetType(CIAPPLISSHEETDO_CIAPPLISSHEETORDO));
		    if (dos == null || dos.Length==0){return null;}
            CiAppLisSheetOrDO[] r = new CiAppLisSheetOrDO[dos.Length];
            for (int i = 0; i < dos.Length;i++ ){
                r[i] = dos[i] as CiAppLisSheetOrDO;
            }
	        return r;
		    
	    }
	    
	    public void setCiAppLisSheetOrDO(CiAppLisSheetOrDO[] dos) {
            if (dos == null || dos.Length == 0) return;
            setChildren(dos[0].GetType(), dos);
	    }


	    public override BaseDO createParentDO()
	    {
	        return new CiAppLisSheetDO();
	    }

        public override BaseDO createChildDO(string clzName)
	    {
            if (CIAPPLISSHEETDO_CIAPPLISSHEETORDO == clzName)
	        {
                return new CiAppLisSheetOrDO();
	        }
	        return null;
	    }
	    
	    /// <summary>
        /// 获取子类全路径名数组
        /// </summary>
        /// <returns></returns>
        public override string[] getChildClassNames(){
            return  new string[]{"iih.ci.ord.app.d.CiAppLisSheetOrDO"};
        }
	    public new static AggMeta GetMeta()
	    {
	        AggMeta meta = new AggMeta();
	        meta.OneToOneDOs = new string[]{};
	        return meta;
	    }
	    
	}
}

File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/ord/app/d/CiapplissheetAggDO.java
package iih.ci.ord.app.d;

import xap.sys.appfw.orm.model.agg.BaseAggDO;
import xap.mw.core.data.BaseDO;
import xap.mw.coreitf.i.IAggDesc;
import xap.mw.coreitf.i.IBaseDO;
import iih.ci.ord.app.d.desc.CiapplissheetAggDODesc;



/**
 * 检验申请单
 */
public class CiapplissheetAggDO extends BaseAggDO {
	private static final long serialVersionUID = 1L;

	@Override
	public CiAppLisSheetDO getParentDO() {
		return ((CiAppLisSheetDO) super.getParentDO());
	}

	public void setParentDO(CiAppLisSheetDO headDO) {
		setParent(headDO);
	}

	public CiAppLisSheetOrDO[] getCiAppLisSheetOrDO() {
		IBaseDO[] dos = getChildren(CiAppLisSheetOrDO.class);
		if(dos==null) return null;
		CiAppLisSheetOrDO[] result=new CiAppLisSheetOrDO[dos.length];
		for(int i=0;i<dos.length;i++){
			result[i]=(CiAppLisSheetOrDO)dos[i];
		}
		return result;
	}
	
	public void setCiAppLisSheetOrDO(CiAppLisSheetOrDO[] dos) {
		setChildren(CiAppLisSheetOrDO.class, dos);
	}

	@Override
	public IAggDesc getAggDesc() {
		CiapplissheetAggDODesc desc = new CiapplissheetAggDODesc();

		return desc;
	}
	
	 public  BaseDO createParentDO() {
         return new CiAppLisSheetDO();
     }

     public  BaseDO createChildDO(String clzName) {
	     if (clzName.equals("iih.ci.ord.app.d.CiAppLisSheetOrDO")) {
             return new CiAppLisSheetOrDO();
         }
         return null;
     }
}